File: song.java
//This program prints a song with a cumulative (or increasing) section

import java.util.*;

class song{
	 
	public static void main(String args[]){
		LinkedList<String> lines = new LinkedList<String>();
		
		//Prints fly stanza, adds lines to increasing section of song
		System.out.println("There was an old woman who swallowed a fly.");
		lines.add("");
		lines.add(0,"Perhaps she'll die.");
		lines.add(0,"I don't know why she swallowed that fly,");
		songPlayer(lines);
		
		//Prints spider stanza, adds line to increasing section of song
		System.out.println("There was an old woman who swallowed a spider,");
		System.out.println("That wriggled and iggled and jiggled inside her.");
		lines.add(0,"She swallowed the spider to catch the fly,");
		songPlayer(lines);
		
		//Prints bird stanza, adds line to increasing section of song
		System.out.println("There was an old woman who swallowed a bird,");
		System.out.println("How absurd to swallow a bird.");
		lines.add(0,"She swallowed the bird to catch the spider,");
		songPlayer(lines);
		
		//Prints cat stanza, adds line to increasing section of song
		System.out.println("There was an old woman who swallowed a cat,");
		System.out.println("Imagine that to swallow a cat.");
		lines.add(0,"She swallowed the cat to catch the bird,");
		songPlayer(lines);
		
		//Prints dog stanza, adds line to increasing section of song
		System.out.println("There was an old woman who swallowed a dog,");
		System.out.println("What a hog to swallow a dog.");
		lines.add(0,"She swallowed the dog to catch the cat,");
		songPlayer(lines);
		
		//Prints bear stanza
		System.out.println("There was an old woman who swallowed a bear,");
		System.out.println("Without a care she swallowed a bear.");
		System.out.println("She swallowed the bear to catch the dog,");
		songPlayer(lines);
		
		//Prints horse stanza
		System.out.println("There was an old woman who swallowed a horse,");
		System.out.println("She died of course.");
	}
	
	//Prints increasing section of song
	public static void songPlayer(LinkedList<String> lines){
		Iterator<String> iter = lines.iterator();
			while(iter.hasNext()){
				System.out.println(iter.next());
			}
	}
}

File: song.py
#Prints a song with a cumulative verse.
cumVer = []

#plays the cumulative lines in order
def verse():
	x = 0
	while x < len(cumVer):
		eval(cumVer[x] + '()')
		x += 1

#prints the lines about the fly with the line from the unique first printed the first pass only.		
def fly():
	if "fly" not in cumVer:
		print("There was an old woman who swallowed a fly")
		cumVer.append("fly")
		verse()
	else:
		print("I don't know why she swallowed the fly,")
		print("Perhaps she'll die!")
		print("")	

#prints the lines about the s[ider with the lines from the unique first printed the first pass only.
def spider():
	if "spider" not in cumVer:
		print("There was an old woman who swallowed a spider,")
		print("That wriggled and iggled and jiggled inside her.")
		cumVer.insert(0, "spider")
		verse()
	else:
		print("She swallowed the spider to catch the fly,")

#prints the lines about the bird with the lines from the unique first printed the first pass only.
def bird():
	if "bird" not in cumVer:
		print("There was an old woman who swallowed a bird,")
		print("How absurd to swallow a bird.")
		cumVer.insert(0, "bird")
		verse()
	else:
		print("She swallowed the bird to catch the spider,")	

#prints the lines about the cat with the lines from the unique first printed the first pass only.
def cat():
	if "cat" not in cumVer:
		print("There was an old woman who swallowed a cat,")
		print("Imagine that to swallow a cat.")
		cumVer.insert(0, "cat")
		verse()
	else:
		print("She swallowed the cat to catch the bird,")

#prints the lines about the dog with the lines from the unique first printed the first pass only.
def dog():
	if "dog" not in cumVer:
		print("There was an old woman who swallowed a dog,")
		print("What a hog to swallow a dog.")
		cumVer.insert(0, "dog")
		verse()
	else:
		print("She swallowed the dog to catch the cat,")

#Prints the entire song.
fly()
spider()
bird()
cat()
dog()

print("There was an old woman who swallowed a bear,")
print("Without a care she swallowed a bear.")
print("She swallowed the bear to catch the dog,")
verse()

print("There was an old woman who swallowed a horse,")
print("She died of course.")